.github/workflows/continuous-build.yml

```diff
@@ -20,49 +20,82 @@ jobs:
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2
+
       - name: Setup Go
         uses: actions/setup-go@v2
         with:
           go-version: '^1.15'
+
       - uses: actions/cache@v2
         with:
           path: ~/go/pkg/mod
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
+
       - name: Run tests
         run: go test -cover ./...
+
       - name: Build daemon
         run: go build ./cmd/tracing
-  build_docker:
-    name: Build Docker image
+  build_publish_daemon_image:
+    name: Build and Publish X-Ray daemon docker image to docker hub and ECR
     needs: build
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2
+
+      - name: Install latest version of aws cli
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
```

Comment:
Can we use the docker image instead of installing the AWS SDK?

https://hub.docker.com/r/amazon/aws-cli

e.g. `docker run --rm amazon/aws-cli ecr-public get-login-password --region us-east-1 | docker login -u AWS --password-stdin public.ecr.aws
`

Reply:
Actually let's use the login action, way simpler

https://github.com/docker/login-action#aws-public-elastic-container-registry-ecr